src/components/ui/Mention.tsx

```diff
@@ -1,42 +1,50 @@
-import React from "react"
+import React, { memo } from "react"
 
 import { getSiteLink } from "~/lib/helpers"
 import { useGetSite } from "~/queries/site"
 
 import { CharacterFloatCard } from "../common/CharacterFloatCard"
 import { UniLink } from "./UniLink"
 
-export const Mention: React.FC<{
-  id: string
-  children?: any
-}> = ({ id, children }) => {
-  let siteId
+const getSiteId = ({ id, children }: { id: string; children?: any }) => {
   if (
     children?.[0] &&
     typeof children[0] === "string" &&
     children[0].startsWith("@")
   ) {
-    siteId = children[0].replace(/^@/, "")
+    return children[0].replace(/^@/, "")
   } else if (id) {
-    siteId = id.replace(/^user-content-/, "")
+    return id.replace(/^user-content-/, "")
   }
+}
+
+export const Mention: React.FC<{
+  id: string
+  children?: any
+}> = memo(
+  function Mention({ id, children }) {
+    let siteId = getSiteId({ id, children })
 
-  const site = useGetSite(siteId)
+    const site = useGetSite(siteId)
 
-  if (siteId && site.data) {
-    return (
-      <CharacterFloatCard siteId={siteId}>
-        <UniLink
-          href={getSiteLink({
-            subdomain: siteId,
-          })}
-          className="inline-block"
-        >
-          @{siteId}
-        </UniLink>
-      </CharacterFloatCard>
-    )
-  } else {
-    return <>{children}</>
-  }
-}
+    if (siteId && site.data) {
+      return (
+        <CharacterFloatCard siteId={siteId}>
+          <UniLink
+            href={getSiteLink({
+              subdomain: siteId,
+            })}
+            className="inline-block"
+          >
+            @{siteId}
+          </UniLink>
+        </CharacterFloatCard>
+      )
+    } else {
+      return <>{children}</>
+    }
+  },
+  (prevProps, nextProps) => {
+    return getSiteId(prevProps) === getSiteId(nextProps)
```

Comment:
Why?

Reply:
`children` is an array of strings, so I think this may be necessary🤔

Reply:
It looks like an attempt to make 10$, but ended up spending an extra 20$. 🤨 (If I missed something, please ignore my confusion.

Reply:
I tried locally and found that changes in children will also trigger updates.

Reply:
> I tried locally and found that changes in children will also trigger updates.

It can only guarantee that the previous content will not re-render when modifying the latter content.

This may be because `rehype-react` uses the index as the key (`h-1`, `h-2` ...) by default for each paragraph node. So when content is added at the beginning of the article, the keys of all subsequent nodes change.